### DISCUSSION TITLE
getSolCast: Rate limit exceeded

### DISCUSSION THREAD

**Discussion by VolkerK62** [StefaE/PVForecast#20] [Q&A]
Hallo,

hat SolCast die Bedingungen geändert?
Ich dachte, es sind 50 Abfragen pro Tag möglich.
AUf der HP steht, für private Anlagen 10 Abrufe pro Tag.

**Comment by StefaE** [comment]
Hallo Volker - in der Tat: auf der HP steht schon lange 10, aber bis jetzt waren immer 50 möglich. Du kannst zwar im config file sagen `apiCalls = 10`, nur funktioniert das nicht bei so kleinen Zahlen. Ich habe eine neue Version die ein anderer Benutzer gerade testet - wenn alles gut geht, mache ich auf Git einen Update Ende Woche.

Sorry for the inconvenience ... :-(

Stefan

**Reply by VolkerK62** [reply]
Hallo Stefan,

kein Problem. Ist mir nur durch Zufall aufgefallen, weil ich mal einen Blick in err.txt geworfen habe.
Da sieht man, dass nach 10 Aufrufen diese RateLimit exceeded Meldung kommt.
In der Config habe ich alles auf default gelassen.

Viele Grüße
Volker

**Comment by StefaE** [comment]
v2.11 handhabt `apiCalls = 10` jetzt korrekt.

**Reply by VolkerK62** [reply]
Danke Stefan.
Aber jetzt habe ich (mal wieder) ein paar dumme Fragen. 
Wie mache ich ein update?
So wie bei der Erstinstallation, nur ohne requirements?
Muss ich die "config.ini" dann neu füllen, oder kann ich die alte weiterverwenden?

**Comment by StefaE** [comment]
Hallo Volker,

- config.ini irgendwo in Sicherheit bringen
- Alle Files in’s gleiche Directory downloaden (entpacken), wo sie schon stehen (für Dich würde schon das überschreiben von solcast.py genügen)
- config.ini wieder dahin schieben wo’s hingehört

Da Du irgendwo einen crontab Eintrag hast, läuft’s dann (hoffentlich :-) ) weiter.
Da Du sowieso nur 10 apiCalls hast, ist das auch der neue Default, und Du musst im config.ini nichts neues einfügen

Stefan
